### DISCUSSION TITLE
Muchas vueltas para usar .pop()

### DISCUSSION THREAD

**Discussion by pipen77** [Taller-Abierto-de-Humanidades-Digitales/curso_python_extenso#36] [Dudas del curso]
Hola Jairo,

Estaba realizando la 3ra parte de la actividad de esta semana y tuve que hacer algunas maromas para poder usar la función .pop y así eliminar el género de cada cadena formada por el catalogo. Este es el el código que hice, pero quisiera saber si existe una forma más elegante para resolver el problema:

```
`# Catalogo

libros = ['Cuentos completos 1; Isaac Asimov; 2009; Ficción', 'Cuentos completos 2; Isaac Asimov; 2009; Ficción', 'De Animales a Dioses; Yuval Noah Harari; 2015; Historia General', 'Homo Deus; Yuval Noah Harari; 2016; Historia General', 'Prohibido Nacer; Trevor Noah; 2017; Autobiografía', 'Somos Luces Abismales; Carolina Sanín; 2018; Ensayos Literarios', 'Muerte con Pingüino; Andrei Kurkov; 1996; Novela', 'Que Viva la Música!; Andrés Caicedo; 1977; Novela']


lis_ani_libros = []
lis_ani_lib_singen = []
lis_lib_singen = []

for libro in libros:
    lis_ani_libros.append(libro.split(';'))
        

for sin_genero in lis_ani_libros:
    sin_genero.pop()
    lis_ani_lib_singen.append(sin_genero)

prelib =[]

for libro in lis_ani_lib_singen:
    prelib.append("; ".join(libro))

libros = prelib

print(libros)`

```

**Comment by pipen77** [comment]
Si no tuviéramos que usar la función .pop que nos obliga a convertir cada elemento de la lista en una nueva cadena usaría esta estructura:

```
for libro in libros:
    lis_ani_libros.append("; ".join(libro.split(';')[0:-1]))
```

Quedo atento a tus comentarios. Gracias.

**Comment by jairomelo** [comment]
Hola @pipen77 

Sí claro, existen muchas maneras de hacer esta tarea eficiente :)  En la actividad solamente complicamos un poco las cosas para reforzar la resolución de problemas, pero efectivamente hay maneras más eficientes de hacer esa tarea. Una de ellas es como la estás proponiendo, e incluso puede hacerse más eficiente si la convertimos en una lista de comprensión:

`libros = ["; ".join(libro.split("; ")[:-1]) for libro in libros] `

De esa manera ya se habrá reemplazado la lista original con los nuevos valores :)

Muchas gracias por tu experimentación 😃
